test/oas3/execute/style-serializer.js

```diff
@@ -37,9 +39,105 @@ describe('OAS3 style serializer', () => {
         '%F0%9F%91%A9%E2%80%8D%F0%9F%91%A9%E2%80%8D%F0%9F%91%A7%E2%80%8D%F0%9F%91%A7'
       );
     });
+  });
+
+  describe('valueEncoder', () => {
+    test('should correctly encode primitive values with escape set to `reserved`', () => {
+      const tested = (value) => valueEncoder(value, 'reserved');
+
+      expect(tested(123)).toEqual('123');
+      expect(tested('a#b$c%d[e]_~1')).toEqual(escape('a#b$c%d[e]_~1'));
+      expect(tested(false)).toEqual('false');
+    });
+
+    test('should correctly encode primitive values with escape set to `unsafe`', () => {
+      const tested = (value) => valueEncoder(value, 'unsafe');
+
+      expect(tested(123)).toEqual('123');
+      expect(tested('a#b$c%d[e]_~1')).toEqual(`a#b$c${escape('%')}d[e]_~1`);
+      expect(tested(false)).toEqual(escape('false'));
+    });
+
+    test('should correctly encode objects with escape set to `reserved`', () => {
+      const tested = (value) => valueEncoder(value, 'reserved');
+
+      expect(tested({ a: 'a#b$c%d[e]_~1' })).toEqual(escape('{"a":"a#b$c%d[e]_~1"}'));
+      expect(tested({ a: { b: { c: 'd' } } })).toEqual(escape('{"a":{"b":{"c":"d"}}}'));
+      expect(tested({ a: 123 })).toEqual(escape('{"a":123}'));
+      expect(tested({ a: false })).toEqual(escape('{"a":false}'));
+    });
+
+    test('should correctly encode objects with escape set to `unsafe`', () => {
+      const tested = (value) => valueEncoder(value, 'unsafe');
+
+      expect(tested({ a: 'a#b$c%d[e]_~1' })).toEqual(
+        `${escape('{"a"')}:${escape('"')}a#b$c${escape('%')}d[e]_~1${escape('"}')}`
```

Comment:
Is it possible to rewrite this to something more sensible? I cannot really read this or tell if this is right or not (as a human).